### ISSUE TITLE
🚀 Feature: Export components that make up Notification page

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#27841]
### 🔖 Feature description

We would like to be able to re-use the components that make up the Notification page - filters and table - as part of a custom plugin. Would a PR to add exports for these be accepted?

### 🎤 Context

We are building a custom plugin and would like to use the Notification components but currently it's all in the page which includes the header. This doesn't work well with what we are trying to build.

### ✌️ Possible Implementation

Export the Notification [filter](https://github.com/backstage/backstage/blob/master/plugins/notifications/src/components/NotificationsFilters/NotificationsFilters.tsx) and [table](https://github.com/backstage/backstage/blob/master/plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx) components

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by Dharmil4602** [comment]
Hi @awanlin, I can take this issue. Please assign it to me

**Comment by awanlin** [comment]
Sure @Dharmil4602, but I'd like to make sure this is something that the Maintainers of the Notification project area are open to first.

**Comment by Dharmil4602** [comment]
Got it

**Comment by awanlin** [comment]
Hi @drodil, @mareklibra, what are your thoughts on this?

**Comment by drodil** [comment]
@awanlin makes sense. Though I would move those to `react` plugin (if they are not yet, can't remember the details) instead as frontend plugin depending on another frontend plugin is not the right way.

**Comment by awanlin** [comment]
Thanks @drodil!

@Dharmil4602, can you run with this based on the above guidance?

**Comment by Dharmil4602** [comment]
Sure, I'll give it a try

**Comment by awanlin** [comment]
Looks like we would need to create the `-react` package and move the components into that.

**Comment by drodil** [comment]
@Dharmil4602 just remember if there is need to move some already exported component from `notifications` to `notifications-react`, remeber to re-export those and mark as deprecated in the `notifications` to not break the API contract right away. I am not sure if there are any cases this would be necessary..

Additionally it seems the `react` plugin is non-existing so you need to create that. I would probably move also the API (ref and type def) to the `react` plugin so it could be used from other plugins but that can be done in another PR if needed.

**Comment by Dharmil4602** [comment]
Got it. I'll make sure of it.
Thanks @drodil @awanlin

**Comment by drodil** [comment]
Noticed that also the settings card could be moved to react plugin. That's at least one component that is exported already from the notifications plugin itself and needs a deprecated re-export.

**Comment by Dharmil4602** [comment]
Got it

**Comment by Dharmil4602** [comment]
@drodil @awanlin I have a question. How to mark component as deprecated?

**Comment by drodil** [comment]
You want to re-export the deprecated components in the `notifications` plugin. So you should add the following to the `notifications/src/index.ts`:

```ts
import { UserNotificationSettingsCard as UserNotificationSettingsCardTemp } from '@backstage/plugin-notifications-react';

/**
 * @public
 * @deprecated use import from {@link @backstage/plugin-notifications-react#UserNotificationSettingsCard} instead as this has now been moved.
 */
export const UserNotificationSettingsCard = UserNotificationSettingsCardTemp;
```

Were there public interfaces you had to move?

**Comment by Dharmil4602** [comment]
Not so far

**Comment by Dharmil4602** [comment]
@drodil I think there is routeRef used in Notifications. If I'll be creating new package, how should I handle the routeRef? What should be done?

**Comment by drodil** [comment]
@Dharmil4602 rhe routeRef should stay in the frontend package as well as the NotificationPage. But other components can be moved to the react plugin
